aiida/orm/nodes/data/singlefile.py

```diff
@@ -56,17 +56,24 @@ def filename(self):
         """
         return self.get_attribute('filename')
 
-    def open(self, key=None, mode='r'):
+    def open(self, path=None, mode='r', key=None):
         """Return an open file handle to the content of this data node.
 
+        :param path: the relative path of the object within the repository.
         :param key: optional key within the repository, by default is the `filename` set in the attributes
```

Comment:
Do you mean to add this in all docstrings of deprecated parameters or just this one? And just the `(deprecated)` or more info such as in which version it was deprecated and when it will be removed

Reply:
Yeah, I think it should go on all deprecated parameters. Not sure if mentioning when it will be removed is necessary - I just want to avoid people looking at the docstring when implementing new stuff to use it.

But yeah, `(deprecated since v1.4, to be removed in v2.0)` is also fine.

Reply:
Is this going to be done in the end?

Reply:
It _is_ already done

Reply:
Shouldn't we also move the `by default is the 'filename' set in the attributes` from `key` to `path`?